packages/server-wallet/src/protocols/__test__/defund-channel.test.ts

```diff
@@ -0,0 +1,175 @@
+import {defaultTestConfig} from '../..';
+import {createLogger} from '../../logger';
+import {DBDefundChannelObjective, ObjectiveModel} from '../../models/objective';
+import {Store} from '../../wallet/store';
+import {testKnex as knex} from '../../../jest/knex-setup-teardown';
+import {seedAlicesSigningWallet} from '../../db/seeds/1_signing_wallet_seeds';
+import {channel} from '../../models/__test__/fixtures/channel';
+import {Channel} from '../../models/channel';
+import {MockChainService} from '../../chain-service';
+import {WalletResponse} from '../../wallet/wallet-response';
+import {ChannelDefunder} from '../defund-channel';
+import {ChallengeStatus} from '../../models/challenge-status';
+import {stateVars} from '../../wallet/__test__/fixtures/state-vars';
+import {stateWithHashSignedBy} from '../../wallet/__test__/fixtures/states';
+import {alice, bob} from '../../wallet/__test__/fixtures/signing-wallets';
+
+const logger = createLogger(defaultTestConfig());
+const timingMetrics = false;
+
+let store: Store;
+beforeEach(async () => {
+  store = new Store(
+    knex,
+    defaultTestConfig().metricsConfiguration.timingMetrics,
+    defaultTestConfig().skipEvmValidation,
+    '0'
+  );
+
+  await store.dbAdmin().truncateDB();
+  await seedAlicesSigningWallet(knex);
+});
+
+test('it should submit a conclude transaction even if there is an active challenge if there is a conclusion proof', async () => {
+  const chainService = new MockChainService();
+  const channelDefunder = ChannelDefunder.create(store, chainService, logger, timingMetrics);
+
+  const c = channel({
+    channelNonce: 1,
+    vars: [stateWithHashSignedBy([alice(), bob()])({isFinal: true})],
+  });
+
+  await Channel.query(knex).withGraphFetched('signingWallet').insert(c);
+
+  const challengeState = {
+    ...c.channelConstants,
+    ...stateVars(),
+  };
+  await ChallengeStatus.insertChallengeStatus(knex, c.channelId, 100, challengeState);
+  const obj = createPendingObjective(c.channelId);
+  await knex.transaction(tx => ObjectiveModel.insert(obj, tx));
+  const response = WalletResponse.initialize();
+  const concludeSpy = jest.spyOn(chainService, 'concludeAndWithdraw');
+
+  await channelDefunder.crank(obj, response);
+
+  const reloadedObjective = await store.getObjective(obj.objectiveId);
```

Comment:
Why doesn't `crank` return the new objective's value?

Reply:
The `ObjectiveManager` assumes a signature of`Promise<void>`. I think it would be fine to change all `crank` methods to return the objective but I think the behaviour should be consistent across all of them.

Also, I think there is value in fetching the objective from the DB to avoid issues like #3122 where we return something different than what we store.

Reply:
💡 `crank` can be a generic function on the `ObjectiveManager`, accepting an objective of type `T`, and returning `Promise<T>`